### PR TITLE
Feature/pdo connection swapping

### PR DESCRIPTION
**Pull request by rhift**
* PDO Connection swapping.
* Fix Exceptions.
* Start to move the `Doctrine\Connection\Decorator\Repository` to ROA.

### CODE REVIEW

File: src/Runtime/ExceptionInterface.php [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+namespace Neighborhoods\Kojo\Runtime;
+
+interface ExceptionInterface extends \Throwable, \JsonSerializable
+{
+    public function setCode(string $code): ExceptionInterface;
+
+    public function addMessage(string $additionalMessage): ExceptionInterface;
+
+    public function jsonSerialize();
```
[the remark this answers is not in the document]
Comment (rhift) [reply src/Runtime/ExceptionInterface.php]: ty <3

**Comment by rhift** [comment]
> Looks good, I just have one high-level question: when a job is scheduled in the middle of a transaction, but that transaction is then rolled back, is there any other state anywhere that needs to be cleaned up (e.g. in redis)?

@p-mucha - really, really good question.

> Or is the scheduling of a job purely the row in the kojo_job table?

Yes, by design, because of your exact and expert insightful concern above.

**Comment by rhift** [comment]
@p-mucha - re-adding you just to make sure that I answered your question well.

**Comment by rhift** [comment]
sorry, gotta merge this in

**Comment by p-mucha** [comment]
@rhift all good, I was just double-checking

**Pull request merged**